recipes-extended/xen/blktap3/remove-creation-of-unused-log-files.patch

```diff
@@ -0,0 +1,150 @@
+################################################################################
+SHORT DESCRIPTION: 
+################################################################################
+Remove creation of unused log files.
+
+################################################################################
+LONG DESCRIPTION: 
+################################################################################
+Remove creation of unused log files.
+
+################################################################################
+CHANGELOG 
+################################################################################
+Author(s):
+Mahantesh Salimath <salimathm@ainfosec.com>
+
+################################################################################
+REMOVAL 
+################################################################################
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+Yes, could be upstreamed.
+
+################################################################################
+INTERNAL DEPENDENCIES 
+################################################################################
+
+################################################################################
+PATCHES 
+################################################################################
+Index: blktap/drivers/tapdisk-log.c
+===================================================================
+--- blktap.orig/drivers/tapdisk-log.c
++++ blktap/drivers/tapdisk-log.c
+@@ -73,74 +73,6 @@ tlog_logfile_vprint(const char *fmt, va_
+ 	tapdisk_logfile_vprintf(&tapdisk_log.logfile, fmt, ap);
+ }
+ 
+-static void __printf(1, 2)
+-tlog_logfile_print(const char *fmt, ...)
+-{
+-	va_list ap;
+-
+-	va_start(ap, fmt);
+-	tlog_logfile_vprint(fmt, ap);
+-	va_end(ap);
+-}
+-
+-#define tlog_info(_fmt, _args ...)					\
+-	tlog_logfile_print("%s: "_fmt, tapdisk_log.ident, ##_args)
+-
+-/**
+- * Closes the log file.
+- *
+- * @param keep if set to true the log file is never removed. NB if an error has
+- * occurred or a USR1 has been received the log is always kept, independently
+- * of whether @keep is set to false.
+- */
+-static void
+-tlog_logfile_close(bool keep)
+-{
+-	td_logfile_t *logfile = &tapdisk_log.logfile;
+-
+-	if (tapdisk_log.precious || tapdisk_log.errors)
+-		keep = true;
+-
+-	tlog_info("closing log, %lu errors", tapdisk_log.errors);
+-
+-	tapdisk_logfile_close(logfile);
+-
+-	if (!keep)
+-		tapdisk_logfile_unlink(logfile);
+-}
+-
+-static int
+-tlog_logfile_open(const char *name, int level)
+-{
+-	td_logfile_t *logfile = &tapdisk_log.logfile;
```

Comment:
To stay closer to upstream, maybe make this code just return when name is NULL?  Something would need to be figured out for the other calls.

Reply:
These functions just create log files where nothing is actually logged. Removing the hunk will create those files and selinux policy needs to be updated. I prefer these files not be created. Let me know if you think otherwise, I can remove the hunk and update selinux policy accordingly.

Reply:
Oh, so it always tries to create the files.  Making it configurable would be better for upstreaming, but this is fine for openxt.